source/github-helpers/index.ts

```diff
@@ -16,25 +16,33 @@ export const getConversationNumber = (): string | undefined => {
 	return undefined;
 };
 
-/**
-Tested on isRepoTree, isBlame, isSingleFile, isEditFile, isSingleCommit, isCommitList, isCompare. Subtly incompatible with isPR
-Example tag content on public repositories: https://github.com/sindresorhus/refined-github/commits/branch-or-commit/even/with/slashes.atom
-Example tag content on private repositories https://github.com/private/private/commits/master.atom?token=AEAXKWNRHXA2XJ2ZWCMGUUN44LM62
-*/
-export const getCurrentBranch = (): string | undefined => {
-	// .last needed for #2799
-	const feedLink = select.last('link[type="application/atom+xml"]');
-	// The feedLink is not available on `isIssue` #3641
-	if (!feedLink) {
+const typesWithCommittish = ['tree', 'blob', 'blame', 'edit', 'commit', 'commits', 'compare'];
+const titleWithCommittish = / at (?<branch>[\w-/]+)( · [\w-]+\/[\w-]+)?$/i;
+export const getCurrentCommittish = (pathname = location.pathname, title = document.title): string | undefined => {
+	if (!pathname.startsWith('/')) {
+		throw new TypeError(`Expected pathname starting with /, got "${pathname}"`);
+	}
+
+	const [, _user, _repo, type, unslashedCommittish] = pathname.split('/');
+	if (!type) { // Root
+		console.trace('Branch could not be determined');
```

Comment:
A dom query attempt could be added here, maybe similar to what's in #4187 

Probably it'll need to be sync due to its usage in a setter of GitHubURL